src/Compiler/Microsoft.CodeAnalysis.Razor.Compiler/src/SourceGenerators/RazorSourceGenerator.RazorProviders.cs

```diff
@@ -42,7 +45,15 @@ public partial class RazorSourceGenerator
                 razorLanguageVersion = RazorLanguageVersion.Latest;
             }
 
-            var razorConfiguration = new RazorConfiguration(razorLanguageVersion, configurationName ?? "default", Extensions: [], UseConsolidatedMvcViews: true);
+            var minimalReferences = references
+                .Where(r => r.Display is { } display && display.EndsWith("Microsoft.AspNetCore.Components.dll", StringComparison.Ordinal))
+                .ToImmutableArray();
+
+            var isComponentParameterSupported = minimalReferences.Length == 0 
+                ? false 
+                : CSharpCompilation.Create("components", references: minimalReferences).HasAddComponentParameter();
```

Comment:
💭 Since this is the source generator, would it make sense to avoid the extra allocation and array copy caused by `ToImmutableArray()`? Something like this would effectively be allocation free until its known that there are any referenecs.

```suggestion
            using var minimalReferences = new PooledArrayBuilder<MetadataReference>(capacity: references.Length);
            foreach (var reference in references)
            {
                if (reference.Display is { } display &&
                    display.EndsWith("Microsoft.AspNetCore.Components.dll", StringComparison.Ordinal))
                {
                    minimalReferences.Add(reference);
                }
            }

            var isComponentParameterSupported = minimalReferences.Count > 0 
                ? CSharpCompilation.Create("components", references: minimalReferences.DrainToImmutable()).HasAddComponentParameter();
                : false;
```